File: assignments/weeksix/sentimental/credit/credit.py
from cs50 import get_int

while True:
    no = get_int("Number: ")
    if no > 1:
        break

one = no % 10
two = int(no % 100 / 10) * 2
two = int(two % 100 / 10) + (two % 10)
three = int(no % 10**3 / 10**2)
four = int(no % 10**4 / 10**3) * 2
four = int(four % 100 / 10) + (four % 10)
five = int(no % 10**5 / 10**4)
six = int(no % 10**6 / 10**5) * 2
six = int(six % 100 / 10) + (six % 10)
seven = int(no % 10**7 / 10**6)
eight = int(no % 10**8 / 10**7) * 2
eight = int(eight % 100 / 10) + (eight % 10)
nine = int(no % 10**9 / 10**8)
ten = int(no % 10**10 / 10**9) * 2
ten = int(ten % 100 / 10) + (ten % 10)
eleven = int(no % 10**11 / 10**10)
twelve = int(no % 10**12 / 10**11) * 2
twelve = int(twelve % 100 / 10) + (twelve % 10)
thirteen = int(no % 10**13 / 10**12)
fourteen = int(no % 10**14 / 10**13) * 2
fourteen = int(fourteen % 100 / 10) + (fourteen % 10)
fifteen = int(no % 10**15 / 10**14)
sixteen = int(no % 10**16 / 10**15) * 2
sixteen = int(sixteen % 100 / 10) + (sixteen % 10)

checksum = one + two + three + four + five + six + seven + eight + \
    nine + ten + eleven + twelve + thirteen + fourteen + fifteen + sixteen

checksum = checksum % 10

visa1 = int(no / 1000000000000)
visa2 = int(no / 1000000000000000)
amex = int(no / 10000000000000)
master = int(no / 100000000000000)


if checksum != 0:
    print("INVALID")
elif visa1 == 4 or visa2 == 4:
    print("VISA")
elif amex == 34 or amex == 37:
    print("AMEX")
elif master >= 51 and master <= 55:
    print("MASTERCARD")
else:
    print("INVALID")

File: assignments/weekone/credit/credit.c
#include <cs50.h>
#include <stdio.h>

long get_number(void);
long check_number(long no);
void check_species(long c, long no);

int main(void)
{
    long no = get_number();
    long c = check_number(no);
    check_species(c, no);
}


long get_number(void)
{
    long no;
    do
    {
        no = get_long("Number: ");
    }
    while (no < 1);
    return no;
}

long check_number(long no)
{
    long a1 = no % 10;
    long b1 = no % 100 / 10 * 2;
    long b11 = (b1 % 100 / 10) + (b1 % 10);

    long a2 = no % 1000 / 100;
    long b2 = no % 10000 / 1000 * 2;
    long b21 = (b2 % 100 / 10) + (b2 % 10);

    long a3 = no % 100000 / 10000;
    long b3 = no % 1000000 / 100000 * 2;
    long b31 = (b3 % 100 / 10) + (b3 % 10);

    long a4 = no % 10000000 / 1000000;
    long b4 = no % 100000000 / 10000000 * 2;
    long b41 = (b4 % 100 / 10) + (b4 % 10);

    long a5 = no % 1000000000 / 100000000;
    long b5 = no % 10000000000 / 1000000000 * 2;
    long b51 = (b5 % 100 / 10) + (b5 % 10);

    long a6 = no % 100000000000 / 10000000000;
    long b6 = no % 1000000000000 / 100000000000 * 2;
    long b61 = (b6 % 100 / 10) + (b6 % 10);

    long a7 = no % 10000000000000 / 1000000000000;
    long b7 = no % 100000000000000 / 10000000000000 * 2;
    long b71 = (b7 % 100 / 10) + (b7 % 10);

    long a8 = no % 1000000000000000 / 100000000000000;
    long b8 = no % 10000000000000000 / 1000000000000000 * 2;
    long b81 = (b8 % 100 / 10) + (b8 % 10);

    long c = b11 + b21 + b31 + b41 + b51 + b61 + b71 + b81 + a1 + a2 + a3 + a4 + a5 + a6 + a7 + a8;
    return c;
}

void check_species(long c, long no)
{
    int visa1 = no / 1000000000000;
    int visa2 = no / 1000000000000000;
    int amex = no / 10000000000000;
    int master = no / 100000000000000;

    if (c % 10 != 0)
    {
        printf("INVALID\n");
    }
    else if ((visa1 == 4) || (visa2 == 4))
    {
        printf("VISA\n");
    }
    else if ((amex == 34) || (amex == 37))
    {
        printf("AMEX\n");
    }
    else if ((master == 51) || (master == 52) || (master == 53) || (master == 54) || (master == 55))
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
